### PR TITLE
Add disable partition endpoints to Admin API

### PR DESCRIPTION
**Pull request by kbatuigas**
Add the following endpoints:
- `GET /cluster/partitions`: [preview](https://deploy-preview-148--redpanda-docs-preview.netlify.app/api/admin-api.html#tag/Partitions/operation/get_cluster_partitions)
- `GET /cluster/partitions/{namespace}/{topic}`: [preview](https://deploy-preview-148--redpanda-docs-preview.netlify.app/api/admin-api.html#tag/Partitions/operation/get_cluster_partitions_topic)
- `POST /cluster/partitions/{namespace}/{topic}`: [preview](https://deploy-preview-148--redpanda-docs-preview.netlify.app/api/admin-api.html#tag/Partitions/operation/post_cluster_partitions_topic)
- `POST /cluster/partitions/{namespace}/{topic}/{partition}`: [preview](https://deploy-preview-148--redpanda-docs-preview.netlify.app/api/admin-api.html#tag/Partitions/operation/post_cluster_partitions_topic_partition)

Merge this PR before merging https://github.com/redpanda-data/docs/pull/113 as this adds Admin API documentation that will be referred to in the new recovery mode doc.

### CODE REVIEW

File: modules/ROOT/attachments/admin-api.yaml [hunk 1]
```diff
@@ -3972,6 +4093,32 @@ components:
           description: node id
           format: int64
       description: Replica assignment
+    cluster_partition:
+      type: object
+      properties:
+        leader_id:
```
Comment (ztlpn) [inline modules/ROOT/attachments/admin-api.yaml]: It turns out we don't actually return leader_id here. My mistake - forgot to delete it from the schema.

File: modules/ROOT/attachments/admin-api.yaml [hunk 2]
```diff
@@ -947,6 +947,127 @@ paths:
...
+              schema: 
+                type: array
+                items:
+                  $ref: '#/components/schemas/cluster_partition'
+      tags: 
+        -  Partitions 
+  /cluster/partitions/{namespace}/{topic}/{partition}:
+    post: 
+      summary: Disable/enable a single partition
+      description: "Disable or enable a single partition. To disable, use `disabled: true` in request body. To enable, use `disabled: false`."
```
Comment (ztlpn) [inline modules/ROOT/attachments/admin-api.yaml]: `disabled: true/false` should probably be `{"disabled": true/false}` to underscore that this is json

File: modules/ROOT/attachments/admin-api.yaml [hunk 3]
```diff
@@ -947,6 +947,127 @@ paths:
...
+          in: path
+          required: true
+          schema: 
+            type: string
+      requestBody: 
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/disable_enable_partitions'
+        required: false
```
Comment (ztlpn) [inline modules/ROOT/attachments/admin-api.yaml]: currently the body is required here

File: modules/ROOT/attachments/admin-api.yaml [hunk 4]
```diff
@@ -947,6 +947,127 @@ paths:
...
+          in: path
+          required: true
+          schema: 
+            type: string
+        - name: topic
+          in: path
+          required: true
+          schema: 
+            type: string
+      requestBody:
```
Comment (ztlpn) [inline modules/ROOT/attachments/admin-api.yaml]: for GET there shouldn't be a body

File: modules/ROOT/attachments/admin-api.yaml [hunk 5]
```diff
@@ -947,6 +947,127 @@ paths:
...
+          required: true
+          schema: 
+            type: integer
+            format: int64
+      requestBody: 
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/disable_enable_partitions'
+        required: false
```
Comment (ztlpn) [inline modules/ROOT/attachments/admin-api.yaml]: same here - currently required

**Review by ztlpn — APPROVED** [review]

**Review by micheleRP — APPROVED** [review]
lgtm

**Pull request merged**